pkg/stanza/fileconsumer/reader.go

```diff
@@ -43,7 +43,7 @@ type Reader struct {
 	generation     int
 	file           *os.File
 	fileAttributes *FileAttributes
-	eof            bool
+	eof            bool `json:"-"` // json tag excludes embedded fields from storage
```

Comment:
My understanding of json encoding is that unexported fields are ignored by josn.Encode - see:
https://pkg.go.dev/encoding/json#Marshal

Struct values encode as JSON objects. Each **exported** struct field becomes a member of the object, using the field name as the object key, unless the field is omitted for one of the reasons given below.

Reply:
You're right of course. I forgot to consider this. Thanks for pointing it out.